src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1518 @@
+     *
+     * @param  s the string to be parsed.
+     * @return the {@code Float16} value represented by the string
+     *         argument.
+     * @throws NullPointerException  if the string is null
+     * @throws NumberFormatException if the string does not contain a
+     *               parsable {@code Float16}.
+     * @see    java.lang.Float#valueOf(String)
+     */
+    public static Float16 valueOf(String s) throws NumberFormatException {
+        // TOOD: adjust precision of parsing if needed
+        return valueOf(Double.parseDouble(s));
+    }
+
+    /**
+     * {@return a {@link Float16} value rounded from the {@link BigDecimal}
+     * argument using the round to nearest rounding policy}
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754.
+     *
+     * @param  v a {@link BigDecimal}
+     */
+    public static Float16 valueOf(BigDecimal v) {
+        return BigDecimalConversion.float16Value(v);
+    }
+
+    private class BigDecimalConversion {
+        /*
+         * Let l = log_2(10).
+         * Then, L < l < L + ulp(L) / 2, that is, L = roundTiesToEven(l).
+         */
+        private static final double L = 3.321928094887362;
+
+        private static final int P_F16 = PRECISION;  // 11
+        private static final int Q_MIN_F16 = MIN_EXPONENT - (P_F16 - 1);  // -24
+        private static final int Q_MAX_F16 = MAX_EXPONENT - (P_F16 - 1);  // 5
+
+        /**
+         * Powers of 10 which can be represented exactly in {@code
+         * Float16}.
+         */
+        private static final Float16[] FLOAT16_10_POW = {
+            Float16.valueOf(1), Float16.valueOf(10), Float16.valueOf(100),
+            Float16.valueOf(1_000), Float16.valueOf(10_000)
+        };
+
+        public static Float16 float16Value(BigDecimal bd) {
+             int scale = bd.scale();
+             BigInteger unscaledValue = bd.unscaledValue();
+
+              if (unscaledValue.abs().compareTo(BigInteger.valueOf(Long.MAX_VALUE)) <= 0) {
+                 long intCompact = unscaledValue.longValue();
+                 Float16 v = Float16.valueOf(intCompact);
+                 if (scale == 0) {
+                     return v;
+                 }
+                 /*
+                  * The discussion for the double case also applies here. That is,
+                  * the following test is precise for all long values, but here
+                  * Long.MAX_VALUE is not an issue.
+                  */
+                 if (v.longValue() == intCompact) {
+                     if (0 < scale && scale < FLOAT16_10_POW.length) {
+                         return Float16.divide(v, FLOAT16_10_POW[scale]);
+                     }
+                     if (0 > scale && scale > -FLOAT16_10_POW.length) {
+                         return Float16.multiply(v, FLOAT16_10_POW[-scale]);
+                     }
+                 }
+             }
```

Comment:
(nit)
Then indents are not "multiples of 4" in the lines of this method's body.